Sources/MessagingPush/UserNotificationsFramework/UserNotificationsFrameworkAdapter.swift

```diff
@@ -56,36 +58,33 @@ class NotificationCenterFrameworkAdapterImpl: NSObject, UNUserNotificationCenter
     }
 
     func newNotificationCenterDelegateSet(_ newDelegate: UNUserNotificationCenterDelegate?) {
-        notificationCenterDelegateProxy.newNotificationCenterDelegateSet(newDelegate)
+        guard let newDelegate = newDelegate else {
+            return
+        }
+
+        notificationCenterDelegateProxy.addPushEventHandler(UNUserNotificationCenterDelegateWrapper(delegate: newDelegate))
     }
 
+    // MARK: UNUserNotificationCenterDelegate functions.
+
     // Functions called by iOS framework, `UserNotifications`. This adapter class simply passes these requests to other code in our SDK where the logic exists.
+    // Convert UserNotifications files into abstracted data types that our SDK understands.
 
     func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
-        let wasClickEventHandled = pushEventListener.onPushAction(PushNotification(notification: response.notification), didClickOnPush: response.didClickOnPush)
-
-        if wasClickEventHandled {
-            // call the completion handler so the customer does not need to.
-            completionHandler()
-        }
+        pushEventHandler.onPushAction(UNNotificationResponseWrapper(response: response), completionHandler: completionHandler)
     }
 
     func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
-        let response = pushEventListener.shouldDisplayPushAppInForeground(PushNotification(notification: notification))
-        guard let shouldShowPush = response else {
-            // push not handled by CIO SDK. Exit early. Another click handler will call the completion handler.
-
-            return
-        }
-
-        if shouldShowPush {
-            if #available(iOS 14.0, *) {
-                completionHandler([.list, .banner, .badge, .sound])
+        pushEventHandler.shouldDisplayPushAppInForeground(UNNotificationWrapper(notification: notification)) { shouldShowPush in
+            if shouldShowPush {
+                if #available(iOS 14.0, *) {
+                    completionHandler([.list, .banner, .badge, .sound])
+                } else {
+                    completionHandler([.badge, .sound])
+                }
             } else {
-                completionHandler([.badge, .sound])
+                completionHandler([])
             }
-        } else {
-            completionHandler([])
         }
```

Comment:
Main part of this file's diff is to move more of this file's logic into the SDK's code. 
This file should be as small and simple as possible since this file is only executed in production or manual QA.